### PR TITLE
Fix CodeQL SM02211 alert - SerializationBinder

### PR DESCRIPTION
**Pull request by sw-joelmut**
#minor
Fixes #
6515
6516
6517
6519
6520

## Description
This PR fixes the CodeQL SM02211 alert related to unsafe `JsonSerializer` `TypeNameHandling` usage.
For these cases, it implements a custom [SerializationBinder](https://www.newtonsoft.com/json/help/html/SerializeSerializationBinder.htm), allowing only the types that are defined in the configuration or related ones.

## Specific Changes
- Adds the `AllowedTypesSerializationBinder` class to allow only types that are defined in the instance.
- Adds tests for the `AllowedTypesSerializationBinder` class.
- Updated `BlobsStorage`, `AzureBlobStorage`, and `CosmosDbPartitionedStorage` classes to implement the binder.
- Updated `BlobsTranscriptStore` setting the `TypeNameHandling` to None.

## Testing
The following image shows the tests passing successfully and the error message when a type isn't allowed by the process.
image
image

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsStorage.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsStorage.cs:7]: This using is not necessary.

File: libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs:7]: Same here.

File: libraries/Microsoft.Bot.Builder/AllowedTypesSerializationBinder.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            // Return Type when it's inside another Type.
+            if (serializedType.IsNested)
+            {
+                if (IsTypeAllowed(serializedType.ReflectedType))
+                {
+                    return true;
+                }
+            }
+
+            // Return Type when it have Interfaces.
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder/AllowedTypesSerializationBinder.cs:224]:
```suggestion
            // Return Type when it has Interfaces.
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Comment by sw-joelmut** [comment]
@ceciliaavila changes applied, thanks

**Review by ceciliaavila — APPROVED** [review]

**Review by erquirogasw — APPROVED** [review]

**Pull request closed without merging**